### PR TITLE
adds the file remover interface and impl for alfresco

### PR DESCRIPTION
**Pull request by j-veiga**

### CODE REVIEW

**Review by allanmckenzie — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file, which follo
 on [Keep a CHANGELOG](http://keepachangelog.com/). This project adheres to
 [Semantic Versioning](http://semver.org/).
 
+## [2.4.0] - 24.06.2019
+
```
Comment (mapingo) [inline CHANGELOG.md]: Normally we put a '## Added' heading above Described additions

File: file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import javax.inject.Inject;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.core.Response;
+
+@ApplicationScoped
+public class AlfrescoFileRemover implements FileRemover {
+
+    @Inject
+    @GlobalValue(key = "alfrescoDeletePath", defaultValue = "/service/slingshot/doclib/action/file/node/workspace/SpacesStore/")
+    String alfrescoDeletePath;
```
Comment (mapingo) [inline file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java]: We have decided to make all injected fields private, anything new should be marked as private

File: file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Inject
+    @GlobalValue(key = "alfrescoDeleteUser")
+    String alfrescoDeleteUser;
```
Comment (mapingo) [inline file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java]: can be private

File: file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Inject
+    AlfrescoRestClient restClient;
```
Comment (mapingo) [inline file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java]: can be private

File: file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public void remove(final String fileId) {
+        try {
+            final Response response = restClient
+                    .delete(format("%s%s", alfrescoDeletePath, fileId), APPLICATION_JSON_TYPE, headersWithUserId(alfrescoDeleteUser));
+            if (response.getStatusInfo().getFamily().compareTo(Response.Status.Family.SUCCESSFUL) != 0) {
+                throw new FileOperationException(format("Error while deleting the document. Code:%d, Reason:%s", response.getStatus(), response.getStatusInfo().getReasonPhrase()));
+            }
+        } catch (ProcessingException e) {
```
Comment (mapingo) [inline file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemover.java]: ProcessingException could be final

File: file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemoverTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AlfrescoFileRemoverTest {
+
+    @InjectMocks
+    AlfrescoFileRemover fileRemover;
```
Comment (mapingo) [inline file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemoverTest.java]: can be private

File: file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemoverTest.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Mock
+    AlfrescoRestClient restClient;
```
Comment (mapingo) [inline file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileRemoverTest.java]: can be private

File: file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileSenderTest.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AlfrescoFileSenderTest {
+
+    @InjectMocks
+    AlfrescoFileSender fileSender;
```
Comment (mapingo) [inline file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileSenderTest.java]: can be private

File: file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileSenderTest.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Mock
+    AlfrescoRestClient restClient;
```
Comment (mapingo) [inline file-alfresco/src/test/java/uk/gov/justice/services/file/alfresco/AlfrescoFileSenderTest.java]: can be private

**Pull request merged**